movemap.h

```diff
@@ -0,0 +1,27 @@
+#ifndef MOVEMAP_H
+#define MOVEMAP_H
+
+#include <vector>
+#include <tuple>
+#include <map>
+
+#include "element.h"
+
+/**
+ * @brief 
+ * 
+ */
+class MoveMap {
+  public:
+  MoveMap(const std::vector<std::tuple<Element, Element, Element>>);
+  int inputCount(const int sphereType = Element::ALL, const int gemType = Element::ALL) const;
+  int outputCount(const int sphereType = Element::ALL, const int gemType = Element::ALL) const;
```

Comment:
Typically you have public in 1 tab, then everything inside it 1 more, eg.:
```suggestion
    MoveMap(const std::vector<std::tuple<Element, Element, Element>>);
    int inputCount(const int sphereType = Element::ALL, const int gemType = Element::ALL) const;
    int outputCount(const int sphereType = Element::ALL, const int gemType = Element::ALL) const;
```